apps/extension/src/hooks/useQuery.tsx

```diff
@@ -4,15 +4,26 @@ import { sanitize } from "dompurify";
 
 interface SanitizedURLSearchParams {
   get(name: string): string | null;
-};
+  getAll(): Record<string, string | null>;
+}
 
-const toSanitized = (urlSearchParams: URLSearchParams): SanitizedURLSearchParams => ({
+const toSanitized = (
+  urlSearchParams: URLSearchParams
+): SanitizedURLSearchParams => ({
   get: (name: string): string | null => {
     const unsanitized = urlSearchParams.get(name);
-    return unsanitized === null ?
-      unsanitized :
-      sanitize(unsanitized);
-  }
+    return unsanitized === null ? unsanitized : sanitize(unsanitized);
+  },
+  getAll: (): Record<string, string | null> => {
+    const values: Record<string, string | null> = {};
+
+    urlSearchParams.forEach((val, key) => {
+      const unsanitized = val;
+      values[key] = unsanitized === null ? unsanitized : sanitize(unsanitized);
```

Comment:
I'm wondering if we should also be sanitizing the key here, since it could be set by the user.

Reply:
I updated this to only return properties that have associated values, and am sanitizing the key as well. Thanks!